src/osd/osd_types.h

```diff
@@ -1110,7 +1110,7 @@ class pool_opts_t {
   }
 
   template<typename T>
-  T value_or(key_t key, const T& default_value) const {
+  T value_or(key_t key, T& default_value) const {
```

Comment:
This doesn't seem right, where was it selecting the wrong signature?

Reply:
@athanatos,
I forgot that '&&' in template functions is "perfect forwarding / catch-all", and not "R-ref".
See https://godbolt.org/z/dMjf77 for what happens.

Now, digging further, I should probably remove this overload altogether.

Reply:
Right, and you'll want to change std::move in the other overload to std::forward.

Reply:
fixed